Kernel/Net/Socket.cpp

```diff
@@ -123,6 +123,14 @@ ErrorOr<void> Socket::setsockopt(int level, int option, Userspace<const void*> u
             return ENOTSUP;
         }
         return {};
+    case SO_DONTROUTE: {
+        if (user_value_size != sizeof(int))
+            return EINVAL;
+        int routing_disabled;
```

Comment:
A slightly more defensive way of writing this could be:
```cpp
int routing_disabled;
if (user_value_size != sizeof(routing_disabled))
    return EINVAL;
```
That way, we reduce the risk of introducing a type size mismatch bug some day in the future.

Reply:
Agreed, changed it to match (in getsockopt as well)